### PR TITLE
Fix NavDropdown alignRight failing typescript check.

### PR DESCRIPTION
**Pull request by xiaohong3**
Fixes issue #5339 
I changed the `NavDropdownProps` to extends from `DropdownProps` instead of `BsPrefixPropsWithChildren`. This way it will include all of Dropdown's props.

### CODE REVIEW

File: src/NavDropdown.tsx [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
 import NavItem from './NavItem';
 import NavLink from './NavLink';
-import {
-  BsPrefixPropsWithChildren,
-  BsPrefixRefForwardingComponent,
-} from './helpers';
+import { BsPrefixRefForwardingComponent } from './helpers';
 
-export interface NavDropdownProps extends BsPrefixPropsWithChildren {
+export interface NavDropdownProps extends DropdownProps {
```
Comment (kyletsang) [inline src/NavDropdown.tsx:9]: We should omit `as` from `DropdownProps` because `NavDropdown` doesn't use it.

**Comment by taion** [comment]
do we need any of these here? https://github.com/react-bootstrap/react-bootstrap/blob/521fd9276c17532d382bc4dedd87844894da36dd/src/DropdownButton.tsx#L10-L11

File: src/NavDropdown.tsx (hunk 1 above)
Comment (kyletsang) [reply src/NavDropdown.tsx:9]: Hmm actually this might be difficult with `BsPrefixRefForwardingComponent`

**Comment by kyletsang** [comment]
> 
> 
> do we need any of these here?
> 

Looks like we needed the `React.HTMLAttributes` to get stuff like `id` to work.  

`React.PropsWithChildren<PropsFromToggle>` to get `'href' | 'size' | 'variant' | 'disabled';`

**Comment by taion** [comment]
okay, let's add that then? @xiaohong3

File: src/NavDropdown.tsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-  BsPrefixRefForwardingComponent,
-} from './helpers';
+import { BsPrefixRefForwardingComponent } from './helpers';
+import { PropsFromToggle } from './DropdownToggle';
 
-export interface NavDropdownProps extends BsPrefixPropsWithChildren {
+export interface NavDropdownProps
+  extends DropdownProps,
+    Omit<React.HTMLAttributes<HTMLElement>, 'onSelect' | 'title'>,
+    React.PropsWithChildren<PropsFromToggle> {
```
Comment (kyletsang) [inline src/NavDropdown.tsx:13]: We can drop `React.PropsWithChildren<PropsFromToggle>` actually. I looked thru the code and these props are not needed. Other than that, looks good!

**Pull request merged**